src/bosh_azure_cpi/lib/cloud/azure/models/vm_cloud_props.rb

```diff
@@ -64,9 +56,65 @@ def initialize(vm_properties, global_azure_config)
         ephemeral_disk_hash['size'],
         ephemeral_disk_hash['type']
       )
+    end
+
+    private
+
+    # In AzureStack, availability sets can only be configured with 1 update domain.
+    # In Azure, the max update domain count of a managed/unmanaged availability set is 5.
+    def _default_update_domain_count(global_azure_config)
+      global_azure_config.environment == ENVIRONMENT_AZURESTACK ? 1 : 5
+    end
+
+    # In AzureStack, availability sets can only be configured with 1 fault domain and 1 update domain.
+    # In Azure, the max fault domain count of an unmanaged availability set is 3;
+    #           the max fault domain count of a managed availability set is 2 in some regions.
+    #           When all regions support 3 fault domains, the default value should be changed to 3.
+    def _default_fault_domain_count(global_azure_config)
+      if global_azure_config.environment == ENVIRONMENT_AZURESTACK
+        1
+      else
+        global_azure_config.use_managed_disks ? 2 : 3
+      end
+    end
+
+    def _parse_load_balancer_config(vm_properties, global_azure_config)
+      if vm_properties[LOAD_BALANCER_KEY].is_a?(Hash)
+        resource_group_name = if vm_properties[LOAD_BALANCER_KEY][RESOURCE_GROUP_NAME_KEY].nil?
```

Comment:
make it simple? 
resource_group_name  = vm_properties[LOAD_BALANCER_KEY][RESOURCE_GROUP_NAME_KEY] || global_azure_config.resource_group_name

Reply:
good suggestion.